.github/workflows/upload_complete_workflow_metrics.py

```
@@ -0,0 +1,985 @@
+        container_data[FIELD_JOB_ID] = job_id
+        container_data[FIELD_WORKFLOW_ID] = str(self.run_id)
+        container_data[FIELD_REPO] = self.repo
+        container_data[FIELD_WORKFLOW_NAME] = self.workflow_name
+        container_data[FIELD_BRANCH] = self.ref_name
+        
+        # Find the "Build image" step ID
+        build_step_id = None
+        steps = job_data.get('steps', [])
+        for step in steps:
+            if 'build' in step.get('name', '').lower() and 'image' in step.get('name', '').lower():
+                build_step_id = f"{job_id}_{step.get('number', 1)}"
```

Ruff (0.13.2)

Local variable `build_step_id` is assigned to but never used

Remove assignment to unused variable `build_step_id`

(F841)